EBNF.Compiler/Compiler.cs

```diff
@@ -203,6 +203,39 @@ public static string[] CompileEbnf(string rawData, string namespaceName)
                         indentLevel++;
                         code.Add($"{new string('\t', 3 + indentLevel)}remainder = remainder.Substring({concatenation.Length - 2});");
                         break;
+                    case '!': // Regex Value
+                        
+                        code.Add($"var regex = new Regex({concatenation.Remove(0, 1)});");
+                        code.Add($"{new string('\t', 3 + indentLevel)}var match = regex.Match(remainder);");
+                        code.Add($"{new string('\t', 3 + indentLevel)}if (match.Success)");
+                        code.Add($"{new string('\t', 3 + indentLevel)}{{");
+                        indentLevel++;
+                        code.Add($"{new string('\t', 3 + indentLevel)}if (remainder.StartsWith(\" \"))");
+                        code.Add($"{new string('\t', 3 + indentLevel)}" + "{");
+                        indentLevel++;
+                        code.Add($"{new string('\t', 3 + indentLevel)}remainder = remainder.Remove(0, match.Groups[0].Value.Length + 2);");
+                        indentLevel--;
+                    code.Add($"{new string('\t', 3 + indentLevel)}" + "}");
+                        code.Add($"{new string('\t', 3 + indentLevel)}else");
+                        code.Add($"{new string('\t', 3 + indentLevel)}" + "{");
```

Comment:
This also requires brace escaping
```suggestion
                        code.Add($"{new string('\t', 3 + indentLevel)}{{");
```